diff --git a/libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp b/libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp
index 789c682..7329348 100644
--- a/libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp
+++ b/libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp
@@ -274,7 +274,8 @@ void NavEKF3_core::setAidingMode()
                  maxLossTime_ms = frontend->posRetryTimeUseVel_ms;
              }
              posAidLossCritical = (imuSampleTime_ms - lastRngBcnPassTime_ms > maxLossTime_ms) &&
-                    (imuSampleTime_ms - lastPosPassTime_ms > maxLossTime_ms);
+                    (imuSampleTime_ms - lastPosPassTime_ms > maxLossTime_ms) &&
+                    (imuSampleTime_ms - lastVelPassTime_ms > maxLossTime_ms);
          }
 
          if (attAidLossCritical) {
