diff --git a/libraries/AP_NavEKF2/AP_NavEKF2.cpp b/libraries/AP_NavEKF2/AP_NavEKF2.cpp
index 9f7f461..d8a66d8 100644
--- a/libraries/AP_NavEKF2/AP_NavEKF2.cpp
+++ b/libraries/AP_NavEKF2/AP_NavEKF2.cpp
@@ -1311,24 +1311,19 @@ uint32_t NavEKF2::getLastPosNorthEastReset(Vector2f &posDelta)
         return 0;
     }
 
+    pos_down_reset_data.last_function_call = imuSampleTime_us / 1000;
     posDelta.zero();
+    uint32_t lastPosReset_ms = 0;
 
-    // Do the conversion to msec in one place
-    uint32_t now_time_ms = imuSampleTime_us / 1000;
-
-    // The last time we switched to the current primary core is the first reset event
-    uint32_t lastPosReset_ms = pos_reset_data.last_primary_change;
 
     // There has been a change in the primary core that the controller has not consumed
     // allow for multiple consumers on the same frame
-    if (pos_reset_data.core_changed || pos_reset_data.last_function_call == now_time_ms) {
+    if (pos_reset_data.core_changed) {
         posDelta = pos_reset_data.core_delta;
+        lastPosReset_ms = pos_down_reset_data.last_primary_change;
         pos_reset_data.core_changed = false;
     }
 
-    // Record last time controller got the position reset
-    pos_reset_data.last_function_call = now_time_ms;
-
     // There has been a reset inside the core since we switched so update the time and delta
     Vector2f tempPosDelta;
     uint32_t lastCorePosReset_ms = core[primary].getLastPosNorthEastReset(tempPosDelta);
