diff --git a/libraries/AP_GPS/AP_GPS.cpp b/libraries/AP_GPS/AP_GPS.cpp
index 79c45f6..4b38273 100644
--- a/libraries/AP_GPS/AP_GPS.cpp
+++ b/libraries/AP_GPS/AP_GPS.cpp
@@ -1489,7 +1489,7 @@ void AP_GPS::calc_blended_state(void)
             temp_time_2 += (double)timing[i].last_message_time_ms * (double)_blend_weights[i];
             float gps_lag_sec = 0;
             get_lag(i, gps_lag_sec);
-            _blended_lag_sec += gps_lag_sec * _blend_weights[i];
+            _blended_lag_sec += gps_lag_sec * _blended_lag_sec;
         }
     }
     timing[GPS_BLENDED_INSTANCE].last_fix_time_ms = (uint32_t)temp_time_1;
