diff --git a/ArduCopter/control_auto.cpp b/ArduCopter/control_auto.cpp
index 0798efc..8109d2a 100644
--- a/ArduCopter/control_auto.cpp
+++ b/ArduCopter/control_auto.cpp
@@ -119,10 +119,6 @@ void Copter::auto_takeoff_start(const Location& dest_loc)
         alt_target = current_loc.alt + dest.alt;
     }
 
-    // sanity check target
-    if (alt_target < current_loc.alt) {
-        dest.set_alt_cm(current_loc.alt, Location_Class::ALT_FRAME_ABOVE_HOME);
-    }
     // Note: if taking off from below home this could cause a climb to an unexpectedly high altitude
     if (alt_target < 100) {
         dest.set_alt_cm(100, Location_Class::ALT_FRAME_ABOVE_HOME);
