diff --git a/libraries/APM_Control/AP_SteerController.cpp b/libraries/APM_Control/AP_SteerController.cpp
index d82ed41..efda79e 100644
--- a/libraries/APM_Control/AP_SteerController.cpp
+++ b/libraries/APM_Control/AP_SteerController.cpp
@@ -145,7 +145,7 @@ int32_t AP_SteerController::get_steering_out_rate(float desired_rate)
     // We do this in earth frame to allow for rover leaning over in hard corners
     float yaw_rate_earth = ToDeg(_ahrs.get_yaw_rate_earth());
     if (_reverse) {
-        yaw_rate_earth *= -1.0f;
+        yaw_rate_earth = wrap_180(180 + yaw_rate_earth);
     }
     float rate_error = (desired_rate - yaw_rate_earth) * scaler;
 	
