diff --git a/APMrover2/system.cpp b/APMrover2/system.cpp
index 3bcc733..1eb01ca 100644
--- a/APMrover2/system.cpp
+++ b/APMrover2/system.cpp
@@ -153,9 +153,6 @@ void Rover::init_ardupilot()
     // ---------------------------
     reset_control_switch();
 
-    // disable safety if requested
-    BoardConfig.init_safety();
-
     // flag that initialisation has completed
     initialised = true;
 }
